src/plugins/gs-flatpak.c

```diff
@@ -2393,6 +2393,19 @@ gs_flatpak_app_install (GsFlatpak *self,
 	return TRUE;
 }
 
+static gboolean
+error_is_ostree_missing_file_obj (GError *error)
+{
+	if (!g_error_matches (error, G_IO_ERROR,
+			      G_IO_ERROR_NOT_FOUND))
+		return FALSE;
+
+	/* XXX: matching the error messages is not ideal but we need to further
+	 * identify the error we are interested in */
+	return (g_strstr_len (error->message, -1, "No such file object") != NULL) ||
+		(g_strstr_len (error->message, -1, "Couldn't find file object") != NULL);
```

Comment:
@cosimoc yeah, I had mentioned that it was an ugly thing to do. If you prefer we can always allow the retrial when the error simply matches, potentially ending up doing a retrial when the cause is not exactly a missing file object, but the downside of doing a retry when the problem may not be solved is not a big issue...